toolchain/check/handle_interface.cpp

```diff
@@ -126,14 +128,34 @@ auto HandleInterfaceDefinitionStart(
   // Enter the interface scope.
   context.scope_stack().Push(interface_decl_id, interface_info.scope_id);
 
-  // TODO: Introduce `Self`.
-
   context.inst_block_stack().Push();
   context.node_stack().Push(parse_node, interface_id);
 
   // We use the arg stack to build the witness table type.
   context.args_type_info_stack().Push();
 
+  // Declare and introduce `Self`.
+  if (!interface_info.is_defined()) {
+    // TODO: Once we support parameterized interfaces, this won't be the right
+    // type. For `interface X(T:! type)`, the type of `Self` is `X(T)`, whereas
+    // this will be simply `X`.
+    auto self_type_id = context.GetTypeIdForTypeConstant(
+        context.constant_values().Get(interface_decl_id));
+
+    // We model `Self` as a symbolic binding whose type is the interface.
+    interface_info.self_param_id = context.AddInst(
```

Comment:
Should import_ref.cpp be population this now too?

Reply:
Added a TODO. While investigating, I found the existing import_ref support for importing `BindName` and `BindSymbolicName` doesn't work, so I added a couple more TODOs there.